src/librustdoc/clean/inline.rs

```diff
@@ -258,10 +258,27 @@ pub(crate) fn build_external_trait(cx: &mut DocContext<'_>, did: DefId) -> clean
 fn build_external_function<'tcx>(cx: &mut DocContext<'tcx>, did: DefId) -> Box<clean::Function> {
     let sig = cx.tcx.fn_sig(did);
 
-    let predicates = cx.tcx.predicates_of(did);
+    let mut late_bound_regions: Vec<_> = cx
+        .tcx
+        .collect_referenced_late_bound_regions(&sig)
```

Comment:
You should just use `sig.bound_vars()`, which gives the kinds and names of the late-bound variables, and avoid any unpredictability. This will avoid sorting, since they are in source code order.

Reply:
Ah, that's helpful! Thanks :)